app/src/main/java/com/example/mylittlepokemons/data/Pokemon.kt

```diff
@@ -0,0 +1,44 @@
+package com.example.mylittlepokemons.data
+
+import androidx.annotation.DrawableRes
+import com.example.mylittlepokemons.R
+
+data class Pokemon(
+    val id: Int,
+    val name: String,
+    val height: Int,
+    val weight: Int,
+    @DrawableRes val spriteRes: Int,
+    val stats: HashMap<String, Int>,
+    val types: List<String>,
+) {
+    companion object {
+        const val POKEMON_ID_KEY = "pokemon_id"
+
+        const val HP_KEY = "hp"
+        const val ATTACK_KEY = "attack"
+        const val DEFENSE_KEY = "defense"
+        const val SPEC_ATTACK_KEY = "special_attack"
+        const val SPEC_DEFENSE_KEY = "special_defense"
+        const val SPEED_KEY = "speed"
+    }
+
+    fun isSpecial() = types.any(SpecialTypes.getElements()::contains)
```

Comment:
нашёл что искал:
```suggestion
    fun isSpecial() = types.intersect(SpecialTypes.getElements).isNotEmpty()
```
Я думаю это поинтереснее выглядит